packages/cozy-stack-client/src/OAuthClient.js

```diff
@@ -0,0 +1,358 @@
+   * @throws {Error} The URL should contain the same state code as the one generated with `client.getAuthCodeURL()`. If not, it will throw an error
+   * @param   {string} pageURL The redirected page URL, containing the state code and the access code
+   * @param   {string} stateCode The state code that was contained in the original URL the user was sent to (see `client.getAuthCodeURL()`)
+   * @returns {string} The access code
+   */
+  getAccessCodeFromURL(pageURL, stateCode) {
+    if (!stateCode) throw new Error('Missing state code')
+
+    const params = new URL(pageURL).searchParams
+    const urlStateCode = params.get('state')
+    const urlAccessCode = params.get('access_code')
+
+    if (stateCode !== urlStateCode)
+      throw new Error('Given state does not match url query state')
+
+    return urlAccessCode
+  }
+
+  /**
+   * Exchanges an access code for an access token. This function does **not** update the client's token.
+   * @throws {NotRegisteredException} When the client doesn't have it's registration information
+   * @param   {string} accessCode The access code contained in the redirection URL — sett `client.getAccessCodeFromURL()`
+   * @returns {Promise} A promise that resolves with an AccessToken object.
+   */
+  async fetchAccessToken(accessCode) {
+    if (!this.isRegistered()) throw new NotRegisteredException()
+
+    const data = new URLSearchParams({
+      grant_type: 'authorization_code',
+      code: accessCode,
+      client_id: this.oauthOptions.clientID,
+      client_secret: this.oauthOptions.clientSecret
+    })
+
+    const result = await this.fetch('POST', '/auth/access_token', data, {
+      headers: { 'Content-Type': 'application/x-www-form-urlencoded' }
+    })
+
+    return new AccessToken(result)
+  }
+
+  /**
+   * Retrieves a new access token by refreshing the currently used token.
+   * @throws {NotRegisteredException} When the client doesn't have it's registration information
+   * @throws {Error} The client should already have an access token to use this function
+   * @returns {Promise} A promise that resolves with a new AccessToen object
+   */
+  async refreshToken() {
+    if (!this.isRegistered()) throw new NotRegisteredException()
+    if (!this.token) throw new Error('No token to refresh')
+
+    const data = new URLSearchParams({
+      grant_type: 'refresh_token',
+      refresh_token: this.token.refreshToken,
+      client_id: this.oauthOptions.clientID,
+      client_secret: this.oauthOptions.clientSecret
+    })
+
+    const result = await this.fetch('POST', '/auth/access_token', data, {
+      headers: { 'Content-Type': 'application/x-www-form-urlencoded' }
+    })
+
+    return new AccessToken({
+      refresh_token: this.token.refreshToken,
+      ...result
+    })
+  }
+
+  /**
+   * Updates the client's stored token
+   * @param {string} token = null The new token to use — can be a string, a json object or an AccessToken instance.
+   */
+  setCredentials(token = null) {
```

Comment:
I checked [mdn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters), as `undefined` or `null` are falsy value, then calling `setCredentials(null)` or `setCredentials(undefined)` will go in the `else` statement, even if there is no default parameter.

> Default function parameters allow formal parameters to be initialized with default values if no value or undefined is passed.

```js
function test(num = 1) {
  console.log(typeof num);
}

test();          // 'number' (num is set to 1)
test(undefined); // 'number' (num is set to 1 too)

// test with other falsy values:
test('');        // 'string' (num is set to '')
test(null);      // 'object' (num is set to null)
```